bin/src/hickory-dns.rs

```diff
@@ -162,90 +164,147 @@ async fn load_zone(
     let zone_name: Name = zone_config
         .zone()
         .map_err(|err| format!("failed to read zone name: {err}"))?;
-    let zone_name_for_signer = zone_name.clone();
     let zone_type: ZoneType = zone_config.zone_type();
-    let is_axfr_allowed = zone_config.is_axfr_allowed();
-    #[allow(unused_variables)]
-    let is_dnssec_enabled = zone_config.is_dnssec_enabled();
-
-    if zone_config.is_update_allowed() {
-        warn!("allow_update is deprecated in [[zones]] section, it belongs in [[zones.stores]]");
-    }
 
     // load the zone and insert any configured authorities in the catalog.
-    debug!(
-        "loading authorities for {zone_name} with stores {:?}",
-        zone_config.stores
-    );
 
     let mut authorities: Vec<Arc<dyn AuthorityObject>> = vec![];
-    for store in &zone_config.stores {
-        let authority: Arc<dyn AuthorityObject> = match store {
-            #[cfg(feature = "sqlite")]
-            StoreConfig::Sqlite(config) => {
-                let mut authority = SqliteAuthority::try_from_config(
+
+    #[cfg(feature = "blocklist")]
+    let handle_blocklist_store = |config| {
+        let zone_name = zone_name.clone();
+
+        async move {
+            Result::<Arc<dyn AuthorityObject>, String>::Ok(Arc::new(
+                BlocklistAuthority::try_from_config(
                     zone_name.clone(),
                     zone_type,
-                    is_axfr_allowed,
-                    is_dnssec_enabled,
-                    Some(zone_dir),
                     config,
-                    #[cfg(feature = "dnssec")]
-                    zone_config.nx_proof_kind.clone(),
+                    Some(zone_dir),
                 )
-                .await?;
+                .await?,
+            ))
+        }
+    };
 
-                // load any keys for the Zone, if it is a dynamic update zone, then keys are required
-                load_keys(&mut authority, zone_name_for_signer.clone(), zone_config).await?;
-                Arc::new(authority)
-            }
-            StoreConfig::File(config) => {
-                let mut authority = FileAuthority::try_from_config(
-                    zone_name.clone(),
-                    zone_type,
-                    is_axfr_allowed,
-                    Some(zone_dir),
-                    config,
-                    #[cfg(feature = "dnssec")]
-                    zone_config.nx_proof_kind.clone(),
-                )?;
+    let handle_default_store = || {
```

Comment:
Nit: rename this to something like `empty_stores_error()`? Also let's move it below `load_zone()` (outside)?